docs/src/dev/future/proposal-transaction-4.asciidoc

```diff
@@ -0,0 +1,115 @@
+////
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+////
+image::apache-tinkerpop-logo.png[width=500,link="https://tinkerpop.apache.org"]
+
+*x.y.z - Proposal 4*
+
+== Tinkerpop native transaction support
+
+=== Introduction
+Now, if you need to use transactions in tinkerpop, the only solution is to use the Neo4J plugin. Unfortunately, this plugin has not been updated for a long time and is only compatible with Neo4J version 3.4, which reached end of life in March, 2020.
+
+=== Motivation
+We would like to propose to implement native Tinkerpop transaction support. Users will be able to use transactions without the outdated Neo4J plugin.
+
+=== Assumptions
+
+* Need to reuse existing code as much as possible.
+* Users should be able to work both with and without transactions.
+* Transaction implementation should not have additional external dependencies.
+* Existing drivers should work without changes.
+
+=== Specifications
+==== Design Overview
+The main changes relate to TinkerGraph, we need to make a similar solution but with transaction support. I think TinkerGraph should be unchanged because for many users performance will be more important.
+I think by default I can make 2 configurations on Gremlin Server and users themselves will choose what they need, TinkerGraph or TinkerTransactionGraph.
```

Comment:
Will users be able to run an embedded TinkerTransactionGraph?

Reply:
yes, both `TinkerTransactionGraph` and `TinkerGraph` should give similar capabilities